src/vs/base/node/id.ts

```diff
@@ -99,3 +100,21 @@ async function getMacMachineId(errorLogger: (error: any) => void): Promise<strin
 		return undefined;
 	}
 }
+
+const SQM_KEY: string = 'Software\\Microsoft\\SQMClient';
+export async function getSqmMachineId(errorLogger: (error: any) => void): Promise<string> {
+	if (isWindows) {
+		const Registry = await import('@vscode/windows-registry');
+		try {
+			// Wait for 1s max (as to not block the startup) to read the SQM value
+			return await Promise.race([
+				Registry.GetStringRegKey('HKEY_LOCAL_MACHINE', SQM_KEY, 'MachineId') || '',
```

Comment:
@isidorn Only now I realise that `Registry.GetStringRegKey` seems to be sync, so this race can be removed because the operation is not async at all.

Reply:
I was thinking about this while brushing my teeth this morning. Will update. Thanks